util/src/main/java/io/kubernetes/client/util/KubeConfig.java

```diff
@@ -61,22 +68,35 @@ public KubeConfig(ArrayList<Object> contexts,
         this.users = users;
     }
 
-    public void setContext(String context) {
+    public boolean setContext(String context) {
         currentCluster = null;
         currentUser = null;
-        currentContext = (Map<String, Object>) findObject(contexts, context).get("context");
+        Map<String, Object> ctx = findObject(contexts, context);
+        if (ctx == null) {
+            System.err.println("Failed to find context: " + context + " in " + contexts);
```

Comment:
I debated this... I decided that I preferered "return false" rather than "throw" because the context not existing isn't "exceptional" per-se but rather a potentially expected outcome...

(and I should probably get rid of the stderr logging)

If you feel strongly about throwing instead of returning false, I'm ok with that, this is a gray area.

(there is some interesting discussion on the subject here: https://codereview.stackexchange.com/questions/11724/is-it-better-practice-to-have-void-method-throw-an-exception-or-to-have-the-meth)